ruby/ql/lib/codeql/ruby/ast/internal/Variable.qll

```diff
@@ -296,10 +307,10 @@ private module Cached {
   }
 
   cached
-  predicate access(Ruby::Identifier access, VariableReal variable) {
+  predicate access(Ruby::AstNode access, VariableReal variable) {
```

Comment:
With this change, I think we should change `access(this, _)` to `access(this.(Ruby::Identifier), _)` in the charpred of `Access` below.

Reply:
That would exclude implicit variables like `x` in `{x:}` , wouldn't it?

Reply:
No, I don't think so. It would exclude the _key_ `x` in `{x:}` from the class `Access`, which should be fine, as you have also adjusted `Variable::isImplicitWrite` to include `any(HashPattern p).getValue(_)`.